ps_landing_page/models/hr_employee_landing_page.py

```diff
@@ -183,11 +202,28 @@ def _default_employee(self):
     )
     current_week = fields.Boolean(compute="_compute_all")
 
+    def last_employement_week(self):
+        date_range = self.env["date.range"]
+        emp_obj = self.env.user.employee_id
+        date_range_type_cw_id = self.env.ref(
+            "ps_date_range_week.date_range_calender_week"
+        ).id
+        employment_date = emp_obj.official_date_of_employment
+        return date_range.search(
+            [
+                ("type_id", "=", date_range_type_cw_id),
+                ("date_start", "<=", employment_date),
+                ("date_end", ">=", employment_date),
+            ]
+        )
+
     def get_unsubmitted_timesheet(self):
+        employment_week = self.last_employement_week()
         return self.env["hr_timesheet.sheet"].search(
             [
                 ("user_id", "=", self.env.user.id),
                 ("state", "in", ("draft", "new")),
+                ("date_start", ">=", employment_week.date_start),
```

Comment:
it's unclear to me which problem this solves, why are there unsubmitted timesheets for employees before their employment date?

Reply:
I might interpret this wrong, but what we are trying to solve is this:
- The landing page shows the oldest timesheet that still needs to be submitted, greater than the employment date
- When clicking the timesheet button, the user should be brought to that timesheet

Reply:
why the restriction greater than the employment date? Will there be employees that come and go, and you want to take care this only looks at their current stay? If not, better keep it simple and just show the earliest unsubmitted timesheet of the employee.

Reply:
Hi Holger, yes we want this to look at the current stay. Example: We have 'flexible' employees that stay for a few years and then go away for a year and then come back on board. In this case we only want to consider the timesheets of their latest stay. Otherwise they would have to fill in several years worth of timesheets.